### DISCUSSION TITLE
Incorrect tab position

### DISCUSSION THREAD

**Discussion by DreamC** [piroor/treestyletab#3179] [Q&A]
Since I installed TST a few months ago I noticed some strange behavior with opened tabs. TST is set to open new child tabs next to recent tab (imitate Firefox behavior). Sometimes new tabs keep on creating after some sort of "old recent" tab even if focus them. Also focusing a tab while it is still loading does nothing - new child tab will be opened after recent tab instead of current.
I tried a fresh profile, but the problem is still there.
Is this the intended behavior or not? And should it be different from the "Firefox controls opened tabs"?

Video that will help to visualize it: https://youtu.be/bm_Aaz3N2nk

**Comment by irvinm** [comment]
I just created a new instance with no changes and installed TST and left it at defaults.

The tabs settings defaulted to:

![image](https://user-images.githubusercontent.com/979729/188004659-46a1511c-46c2-4644-a641-dd439a4dea31.png)

When I tried your scenario, I created a tab and then from that tab opened links from that page as children and each of those new children were always created after the last child that was created.  I then clicked on a few of the children and went to the partent and created more children tabs and again each tab (different that what you were seeing) were still created at the bottom of the tree.  (FF 104.0.1, Win10, TST 3.8.26)

https://user-images.githubusercontent.com/979729/188005586-60ec1ff9-c3ef-4280-9c9f-f56d7ee8542b.mp4

**Reply by DreamC** [reply]
Your first screenshot is about new child tabs opened from PINNED tabs.
The correct setting is in the "Tree behavior" group at the very end.

Try opening a child tab and while it is still loading (fav icon is yet not shown), change focus to it and back. Now open another child tab.

**Reply by irvinm** [reply]
> Your first screenshot is about new child tabs opened from PINNED tabs. The correct setting is in the "Tree behavior" group at the very end.
> 
Fair enough!  There are a lot of options. :)

> Try opening a child tab and while it is still loading (fav icon is yet not shown), change focus to it and back. Now open another child tab.

I still couldn't change the behavior ... did I miss anything?

https://user-images.githubusercontent.com/979729/188045596-e82acd37-6cb2-4f85-9ac0-efdc1dc6fdf2.mp4

**Reply by piroor** [reply]
Hmm, it may be triggered due to failure of updating `lastRelatedTab` around the code:

https://github.com/piroor/treestyletab/blob/d6c455bd31a295f89abdc4d0f37966a3be659333/webextensions/background/successor-tab.js#L184-L220

This looks need to be tracked as an issue instead of a discussion (Q&A).

**Reply by DreamC** [reply]
Well, I didn't know if this actually was an issue or I was doing something wrong.

But anyway, what is the difference between the "Firefox controls" and "Emulate" options? Or the only point should be that the first one allows other addons to work?

**Reply by konradmb** [reply]
> Your first screenshot is about new child tabs opened from PINNED tabs. The correct setting is in the "Tree behavior" group at the very end.

This is very confusing. The correct setting should be moved to "Tabs opened from Existing Tabs", because that is exactly what it does. I've never found this other setting in "Tree behaviour".

I've uninstalled this addon the first time because of the way it messes with new tab creation and moved to Sidebery. Only now I've discovered that this behaviour actually can be changed. :raised_eyebrow:

**Reply by DreamC** [reply]
Yes, it was confusing for me as well why this option is in that section. Couldn't stick with Sideberry (even though it usually performs smoother) mostly because of interface - TST does great job at mimicking FF.

**Reply by konradmb** [reply]
Yes, the interface is nearly native. The right click menu shows all of the options coming from extensions, which is what I was missing in Sideberry.

**Comment by piroor** [comment]
This is tracked at #3180 and maybe fixed with recent commits.
